packages/plugins/RedPacket/src/types.ts

```diff
@@ -1,4 +1,43 @@
+import type { NetworkType } from "@masknet/web3-shared-evm";
+
 export enum NFTSelectOption {
     All = 'All',
     Partial = 'Partial',
 }
+
+export enum RequirementType {
+    Follow = 'Follow',
+    Like = 'Like',
+    Repost = 'Repost',
+    Comment = 'Comment',
+    NFTHolder = 'NFTHolder'
+}
+
+
+export type FireflyRedpacketSettings = {
```

Comment:
```ts
export type FireflyRedpacketSettings = { type: RequirementType.Follow } ... | { type: RequirementType.NFTHolder, contractAddress: string }
```

Reply:
For ease of use as a UI state it's easier to just use it as an array type. Eventually, it will just be integrated into the payload type at request.